src/main/java/com/aws/greengrass/lifecyclemanager/UpdateSystemPolicyService.java

```diff
@@ -116,7 +117,8 @@ public boolean discardPendingUpdateAction(String tag) {
         }
         // Signal components that they can resume their work since the update is not going to happen
         lifecycleIPCAgent.sendPostComponentUpdateEvent(new PostComponentUpdateEvent());
-        return pendingActions.remove(tag) != null;
+        pendingActions.remove(tag);
+        return true;
     }
```

Comment:
What happens is the action is in progress, like mentioned in the javadoc?

Reply:
It will return false on line 116